aip/auth/4117.md

```diff
@@ -375,10 +374,128 @@ The auth libraries and applications **must** follow the steps below:
+| success                   | boolean | The status of the response. True in this case.                                                                                                                              |
+| token_type                | string  | The 3rd party subject token type. Must be *urn:ietf:params:oauth:token-type:jwt*, *urn:ietf:params:oauth:token-type:id_token*, or *urn:ietf:params:oauth:token-type:saml2*. |
+| id_token OR saml_response | string  | The 3rd party OIDC token or SAML response.                                                                                                                                  |
+| expiration_time           | number  | The 3rd party subject token expiration time in seconds (unix epoch time).                                                                                                   |
+
+A sample successful executable OIDC response:
+```json
+{
+  "version": 1,
+  "success": true,
+  "token_type": "urn:ietf:params:oauth:token-type:id_token",
+  "id_token": "...",
+  "expiration_time": 1620499962
+}
+```
+
+A sample successful executable SAML response:
+```json
+{
+  "version": 1,
+  "success": true,
+  "token_type": "urn:ietf:params:oauth:token-type:saml2",
+  "saml_response": "...",
+  "expiration_time": 1620499962
+}
+```
+
+Error responses:
+
+| Field Name | Type    | Description                                                            |        
+|------------|---------|:-----------------------------------------------------------------------|
+| version    | number  | The version of the JSON output. Currently only version 1 is supported. |
+| success    | boolean | The status of the response. False in this case.                        |                                                                                                                                                                  |
+| code       | string  | The error code.                                                        |
+| message    | string  | The error message.                                                     |
+
+A sample executable error response:
+```json
+{
+  "version": 1,
+  "success": false,
+  "code": "401",
+  "message": "Caller not authorized."
+}
+```
+
+The auth libraries and applications **must** follow the steps below:
+
+- Check **credential_source** has an **executable** field and no **environment_id**,
+  otherwise skip the rest of the steps.
+- Retrieve the external credential's executable information from the
+  **credential_source.executable** field.
+- Check that the `GOOGLE_EXTERNAL_ACCOUNT_ALLOW_EXECUTABLES` environment variable is not set to **1**. If not, error out.
+- Before running the command, check if **credential_source.executable.output_file** was specified in the credential configuration.
```

Comment:
"Before running the command," looks unnecessary, it is basically "before the next step"